### PR TITLE
Fix `PatrolFinder.which()` so that it takes previous finder(s) into account

### LINKED ISSUE #1150: `PatrolFinder.which()` ignores the finder it's called on
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/leancodepl/patrol/issues)

### Steps to reproduce

It's a quick question. [Source code of which](https://github.com/leancodepl/patrol/blob/b606c7bc3c031a5acce4abcdbdc7f9f121454ced/packages/patrol/lib/src/custom_finders/patrol_finder.dart#L307-L318) seems to ignore previously constructed finder (as it doesn't use `finder` field). If it is intended then it'll be better to add it to doc comment.

### Actual results

.

### Logs

<details>
<summary>Logs</summary>

```
<!-- Replace this line with your logs. Do not remove the backticks! -->
```

</details>

### Patrol version

master

### Patrol Doctor output

<details>
<summary>Patrol Doctor output</summary>

```
<!-- Replace this line with your logs. Do not remove the backticks! -->
```

</details>

### Flutter Doctor output

<details>
<summary>Flutter Doctor output</summary>

```
<!-- Replace this line with your logs. Do not remove the backticks! -->
```

</details>

### PR DESCRIPTION
**Pull request by jBorkowska**
This PR fixes #1150.

### CODE REVIEW

**Comment by bartekpacia** [comment]
Oh nice! Thanks:)

- I believe this fixes #1150
- Some tests are failing
- Add a few new tests to make sure the changes work

File: packages/patrol/test/patrol_finder_test.dart [hunk 1]
```diff
@@ -858,6 +870,37 @@ void main() {
 
         expect($('count: 10'), findsOneWidget);
       });
+
+      patrolTest('finds 2 buttons by its inactive status', ($) async {
```
Comment (bartekpacia) [inline packages/patrol/test/patrol_finder_test.dart:874]:
```suggestion
      patrolTest('finds 2 buttons by their inactive status', ($) async {
```

**Review by bartekpacia — APPROVED** [review]
LGTM.

I wonder though if existing tests can break because of this change (because they depended on the buggy `which()` implementation).

**Comment by jBorkowska** [comment]
> LGTM.
> 
> I wonder though if existing tests in customer apps' can break because of this change (because they depended on the buggy `which()` implementation).

I think not, if they were using it as it was expected.

**Pull request closed without merging**